ui/src/app/components/help-sidebar.tsx

```diff
@@ -103,17 +124,80 @@ interface Props {
 
 interface State {
   activeIcon: string;
+  filteredContent: any;
   sidebarOpen: boolean;
   searchTerm: string;
 }
 export class HelpSidebar extends React.Component<Props, State> {
+  debounceInput: Function;
   constructor(props: Props) {
     super(props);
     this.state = {
       activeIcon: undefined,
+      filteredContent: undefined,
       sidebarOpen: false,
-      searchTerm: undefined
+      searchTerm: ''
     };
+    this.debounceInput = fp.debounce(300, (input: string) => {
+      if (input === '' || input.length < 3) {
+        this.setState({filteredContent: undefined});
+      } else {
+        this.searchHelpTips(input.trim().toLowerCase());
+      }
+    });
+  }
+
+  searchHelpTips(input: string) {
+    // For each object, we check the title first. If it matches, we return the entire content array.
+    // If the title doesn't match, we check each element of the content array for matches
+    const filteredContent = fp.values(JSON.parse(JSON.stringify(sidebarContent))).reduce((acc, section) => {
+      const inputMatch = (text: string) => text.toLowerCase().includes(input);
+      const content = section.reduce((ac, item) => {
+        if (!inputMatch(item.title)) {
+          item.content = item.content.reduce((a, ic) => {
+            if (typeof ic === 'string') {
+              if (inputMatch(ic)) {
+                a.push(ic);
+              }
+            } else if (inputMatch(ic.title)) {
+              a.push(ic);
+            } else {
+              ic.content = ic.content.filter(inputMatch);
+              if (ic.content.length) {
+                a.push(ic);
+              }
+            }
+            return a;
+          }, []);
+        }
+        return item.content.length ? [...ac, item] : ac;
+      }, []);
+      return [...acc, ...content];
+    }, []);
+    this.setState({filteredContent});
+  }
+
+  highlightSearchTerm(stringToHighlight: string) {
```

Comment:
I wonder if we could/should have a utility function to highlight search terms. 

How reusable is this? Could we make it a utility function?

Reply:
Yeah, theoretically we're going to do a search on institute in profile & signup, I could definitely see this being reused

Reply:
Yeah, I considered pulling the concept search highlight function out as a utility and using that, but we're searching differently. Here we're matching on the complete search string and in concept search, we're matching on each individual word in the search string.

Reply:
Moved to highlight function to `utils/index.tsx`